streams/src/main/java/org/apache/kafka/streams/state/internals/StreamThreadStateStoreProvider.java

```diff
@@ -40,6 +41,9 @@ public StreamThreadStateStoreProvider(final StreamThread streamThread) {
     @SuppressWarnings("unchecked")
     @Override
     public <T> List<T> stores(final String storeName, final QueryableStoreType<T> queryableStoreType) {
+        if (streamThread.state() == StreamThread.State.DEAD) {
```

Comment:
@enothereska  the problem is that NOT_RUNNING is overloaded. It means that the thread never started or that the thread is dead. it is better to introduce a new state of `DEAD` so we can distinguish between them. It also might be good to rename `NOT_RUNNING` to something else, like `NOT_STARTED` or something better